packages/devtools/src/ts/devtoolsPanel/Visualization.tsx

```diff
@@ -1,43 +1,15 @@
-import React, { useEffect, useRef, FC } from "react"
-import {
-  DataSet,
-  Network,
-  NodeOptions,
-  EdgeOptions,
-} from "vis-network/standalone"
-import type { DebugTag } from "rxjs-traces"
+import React, { FC, useEffect, useRef } from "react"
+import { Network } from "vis-network/standalone"
 import "./Visualization.css"
-
-interface Node extends NodeOptions {
-  id: string
-}
-interface Edge extends EdgeOptions {
-  id: string
-  from: string
-  to: string
-}
-
-const nodeColors = {
-  filterHit: {
-    active: "#fc9797",
-    inactive: "#fccaca",
-  },
-  filterMiss: {
-    active: "#97c2fc",
-    inactive: "#cadffc",
-  },
-}
+import { edges, nodes, filter$ } from "./visualizationState"
 
 const noop = () => void 0
 export const Visualization: FC<{
-  tags: Record<string, DebugTag>
   filter: string
   onSelectNode?: (id: string, x: number, y: number) => void
   onDeselectNode?: (id: string) => void
-}> = ({ tags, filter, onSelectNode = noop, onDeselectNode = noop }) => {
+}> = ({ filter, onSelectNode = noop, onDeselectNode = noop }) => {
   const container = useRef<HTMLDivElement | null>(null)
-  const nodes = useRef<DataSet<Node>>(new DataSet())
-  const edges = useRef<DataSet<Edge>>(new DataSet())
   const network = useRef<Network | null>(null)
```

Comment:
`nodes` and `edges` are `DataSet`s: A datastructure from `vis-network` that is used to push changes to the graph. I decided to move them out and make the integration by directly using streams